File: Data Structures & Algorithms/max-water-container/submission-5.java
class Solution {
    public int maxArea(int[] heights) {
        int max = Integer.MIN_VALUE;
        int i = 0;
        int j = heights.length-1;
        int area;
        while(i<j){
            area = (j-i) * Math.min(heights[i], heights[j]);
            if(area > max){
                max = area;
            }
            if(heights[i] > heights[j]){
                j--;
            }
            else {
                i++;
            }
        }
        return max;
    }
}

File: Data Structures & Algorithms/is-anagram/submission-5.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.length() != t.length()) return false;
        vector<int> al(26, 0);
        for(int i = 0; i < s.length(); ++i){
            char sc = s[i];
            char tc = t[i];
            al[sc-'a']++;
            al[tc-'a']--;
        }
        for(int i : al){
            if(i != 0) return false;
        }
        return true;
    }
};
